src/Client/Adapter/Socket.php

```diff
@@ -597,8 +612,16 @@ protected function _checkSocketReadTimeout()
             $timedout = $info['timed_out'];
             if ($timedout) {
                 $this->close();
+                if (isset($this->config['executetimeout'])) {
```

Comment:
As commented for the curl adapter, even in this case we should continue to use the `timeout` parameter instead of adding a new one (`executetimeout`).